packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -849,6 +850,12 @@ export function diffProperties(
         // to update this element.
         updatePayload = [];
       }
+    } else if (
+      typeof nextProp === 'object' &&
+      nextProp !== null &&
+      nextProp.$$typeof === REACT_OPAQUE_OBJECT_TYPE
+    ) {
+      nextProp.toString();
```

Comment:
What is this call for? Trigger a warning? Or I thought that warning only fires during render?

Reply:
No, the object only shows up during hydration. this function will set state and then throw an error, which causes React to synchronously retry the render, updating the object from a hydration object to something that resembles a string.

Reply:
May benefit frmo an inline comment explaining the significance.